packages/vc-export/src/KiltAttestationProofV1.ts

```diff
@@ -0,0 +1,668 @@
+      `Credential not matching on-chain data: issuer "${attester}", CType: "${onChainCType}"`
+    )
+  }
+  // 16. Check issuance timestamp
+  const tIssuance = new Date(credential.issuanceDate).getTime()
+  // Accept exact matches as well as timestamps rounded to 1-second precision
+  if (
+    !(timestamp === tIssuance || Math.round(timestamp / 1000) === tIssuance)
+  ) {
+    throw new SDKErrors.CredentialUnverifiableError(
+      `block time ${new Date(
+        timestamp
+      ).toISOString()} does not match issuedAt (${credential.issuanceDate})`
+    )
+  }
+  // 17. + 18. validate federatedTrustModel items
+  const { federatedTrustModel = [] } = credential
+  await Promise.all(
+    federatedTrustModel.map(async (i) => {
+      switch (i.type) {
+        case KILT_ATTESTER_DELEGATION_V1_TYPE: {
+          // make sure on-chain delegation matches delegation on credential
+          if (typeof delegationId === 'string') {
+            const credentialDelegationId = delegationIdFromAttesterDelegation(i)
+            if (u8aEq(credentialDelegationId, delegationId)) {
+              if (i.delegators) {
+                await findDelegators(delegationId, new Set(i.delegators))
+              }
+              return
+            }
+          }
+          throw new SDKErrors.CredentialUnverifiableError(
+            `Delegation ${i.id} does not match on-chain records (${delegationId})`
+          )
+        }
+        case KILT_ATTESTER_LEGITIMATION_V1_TYPE: {
+          const legitimation = i.verifiableCredential
+          if (legitimation) {
+            await verifyProof(
+              legitimation,
+              legitimation.proof as KiltAttestationProofV1,
+              { api }
+            ).catch((cause) => {
+              throw new SDKErrors.CredentialUnverifiableError(
+                `failed to verify legitimation ${i.id}`,
+                {
+                  cause,
+                }
+              )
+            })
+          }
+          break
+        }
+        default: {
+          throw new CredentialMalformedError(
+            `unknown type ${
+              (i as { type: string }).type
+            } in federatedTrustModel`
+          )
+        }
+      }
+    })
+  )
+}
+
+/**
+ * Helps with producing a derivative proof for selective disclosure of claims in credentialSubject.
+ *
+ * @param credentialInput The original verifiable credential.
+ * @param proofInput The original proof.
+ * @param disclosedClaims An array of claims that are to be revealed. The `id` of the credentialSubject is always revealed.
+ * @returns A copy of the `credential` (without proof) where `credentialSubject` contains only selected claims and a copy of `proof` containing only `salt` entries for these.
+ * @example
+ * ```
+ * const { proof, credential } = deriveProof(
+ *  originalCredential,
+ *  originalProof,
+ *  ['name', 'address']
+ * )
+ * const derivedCredential = { ...credential, proof }
+ * ```
+ */
+export function deriveProof(
```

Comment:
I would rename this function so that it is not super confusing with the other ones, particularly `createProof`. Maybe something like `createPresentation`?

Reply:
`createPresentation` would be _very_ confusing, given that creating a presentation for a VC usually means creating (and signing) a Verifiable Presentation. How about `updateProof` ?

Reply:
I would say that `update` is also misleading, as it returns a new object and does not change the input one.
When would this function be called outside of presentations?

Reply:
Practically never, but the result is not a presentation. We could also call it `selectiveDisclosure`? Guess that's closest to what it does.

Reply:
I think the term proof is then the most accurate, since that's what it is, indeed, which is then used in presentations. So I guess `deriveSelectiveDisclosureProof` would work, not sure if it's too long. Also `deriveSDProof` might work.